02week/ticTacToe.js

```diff
@@ -72,22 +69,18 @@ function ticTacToe(row, column) {
   if (validValue(row) && validValue(column)) {  // This test makes sure values entered are 0, 1, 2 and nothing else.
     if (!board[row][column].trim() ) {  // This test makes sure the square is empty.
       board[row][column] = playerTurn;  // set the square equal to the current player.
-
+      turnCount++;
       if (checkForWin()) {  // checkForWin returns TRUE if someone won.  It returns FALSE if no one has won yet.
         console.log(`Yes we have a winner folks... player ${playerTurn}.  Start a new game`);
         return true;  // returning true ends the game.  We have a winner.
-      } else if (checkForTie()) {
+      } else if (turnCount === 9) {
         printBoard();  // refreshes the screen and shows the last play prior to the message below.
         console.log(`Yes we have a tie folks.  No one won.  Start a new game`);
         return true;  // returning true ends the game.  We have a winner.
       } else {
         // Ok no one has won yet.
         // this logic controls who's turn it is.
-        if (playerTurn === 'X') {
-          playerTurn = 'O';
-        } else {
-          playerTurn = 'X';
-        }
+        playerTurn === 'X'? playerTurn = 'O' : playerTurn = 'X';
```

Comment:
You could also write it like this:

`playerTurn = (playerTurn === 'X') ? 'O' : 'X';`

But the one you wrote should work fine.